lib/buffer.js

```diff
@@ -59,18 +59,16 @@ Buffer.prototype.swap32 = function swap32() {
   return swap32n.apply(this);
 };
 
-const flags = bindingObj.flags;
-const kNoZeroFill = 0;
+// |binding.zeroFill| can be undefined when running inside an isolate where we
+// do not own the ArrayBuffer allocator.  Zero fill is always on in that case.
+const zeroFill = bindingObj.zeroFill || [0];
 
 function createBuffer(size, noZeroFill) {
-  flags[kNoZeroFill] = noZeroFill ? 1 : 0;
-  try {
-    const ui8 = new Uint8Array(size);
-    Object.setPrototypeOf(ui8, Buffer.prototype);
-    return ui8;
-  } finally {
-    flags[kNoZeroFill] = 0;
-  }
+  if (noZeroFill)
+    zeroFill[0] = 0;  // Reset by the runtime.
```

Comment:
You mean the constant?  No real reason except that it's a single-element array now.